### ISSUE TITLE
Code Review?

### ISSUE THREAD

**Issue by jbenet** [jbenet/ipfs-paste#1]
@chriscool your sh-fu is much stronger than mine-- care to take a look when you have a chance?

**Comment by chriscool** [comment]
Ok, will have a look.

**Comment by chriscool** [comment]
Done in PR #2 and PR #3.

**Comment by chriscool** [comment]
A few points to discuss:
- I am not sure it can work if the daemon is not launched, maybe we should check that it is launched,
- we could try to open the URL in a browser tab like `git help -w` does, see also http://git-scm.com/docs/git-web--browse.html

**Comment by jbenet** [comment]
> I am not sure it can work if the daemon is not launched, maybe we should check that it is launched,

yeah it can't. we need a good way for programs to deal with these use cases:
1. ipfs service is running locally, use it.
2. ipfs service is running locally, but i want to use another. (maybe ephemeral node).
3. ipfs service is not running. start it, run the cmd, and shut it down after.
4. ipfs service is not running, error out and ask the user to turn it on.

ideally all of this from the `ipfs` cli tool. (may not be in go-ipfs, we could install a wrapper or something)

cc @whyrusleeping

> we could try to open the URL in a browser tab

yeah that might work!
